### PR TITLE
feat: add TF to manage a sensor mig

### PR DESCRIPTION
**Pull request by retr0h**
Initial terraform to deploy a Corelight machine image based Sensor into a Managed Instance Group (MIG) behind the internal load balancer.

### CODE REVIEW

File: modules/bastion/outputs.tf
Comment (thathaneydude) [inline modules/bastion/outputs.tf:1]: Can you output the instance id here?

File: modules/sensor/instance.tf [hunk 1]
```diff
@@ -0,0 +1,62 @@
+resource "google_compute_instance_template" "sensor_template" {
+  name         = "${var.name_prefix}-mig-template"
```
Comment (thathaneydude) [inline modules/sensor/instance.tf:2]: The entire name should be a variable so folks can configure it to their own internal naming standards. Same for all the resources really.

File: modules/sensor/instance.tf [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    subnetwork = var.network_mon_subnet_name
+  }
+
+  lifecycle {
+    create_before_destroy = false
+  }
+}
+
+resource "google_compute_region_instance_group_manager" "sensor_mig" {
+  name               = "${var.name_prefix}-mig-manager"
```
Comment (thathaneydude) [inline modules/sensor/instance.tf:34]: Full name should be a variable

File: modules/sensor/instance.tf [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+  }
+
+  auto_healing_policies {
+    health_check      = google_compute_region_health_check.traffic_mon_health_check.id
+    initial_delay_sec = 600
+  }
+}
+
+resource "google_compute_region_autoscaler" "sensor_autoscaler" {
+  name   = "${var.name_prefix}-autoscale"
```
Comment (thathaneydude) [inline modules/sensor/instance.tf:50]: Same here.

File: modules/sensor/load_balancer.tf [hunk 4]
```diff
@@ -0,0 +1,81 @@
+resource "google_compute_region_health_check" "traffic_mon_health_check" {
+  name               = "${var.name_prefix}-traffic-monitor-health-check"
```
Comment (thathaneydude) [inline modules/sensor/load_balancer.tf:2]: Should be a variable

File: modules/sensor/load_balancer.tf [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+  http_health_check {
+    port         = var.health_check_http_port
+    request_path = "/api/system/healthcheck"
+    response     = "{\"message\":\"OK\"}"
+  }
+}
+
+# allow all access from health check ranges
+resource "google_compute_firewall" "sensor_health_check_rule" {
+  name          = "${var.name_prefix}-sensor-health-check-rule"
```
Comment (thathaneydude) [inline modules/sensor/load_balancer.tf:17]: Should be a variable

File: modules/sensor/load_balancer.tf [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+  allow {
+    protocol = "tcp"
+    ports    = [var.health_check_http_port]
+  }
+
+  target_tags = ["sensor"]
+}
+
+resource "google_compute_region_backend_service" "traffic_ilb_backend_service" {
+  name                  = "${var.name_prefix}-traffic-ilb-backend-service"
```
Comment (thathaneydude) [inline modules/sensor/load_balancer.tf:31]: Should be a variable

File: modules/sensor/outputs.tf
Comment (thathaneydude) [inline modules/sensor/outputs.tf:1]: This should output the ids of the resources deployed

File: modules/sensor/load_balancer.tf [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+  load_balancing_scheme = "INTERNAL"
+  session_affinity      = "NONE"
+
+  backend {
+    group = google_compute_region_instance_group_manager.sensor_mig.instance_group
+  }
+}
+
+resource "google_compute_forwarding_rule" "traffic_forwarding_rule" {
+  name                   = "${var.name_prefix}-traffic-forwarding-rule"
```
Comment (thathaneydude) [inline modules/sensor/load_balancer.tf:45]: Should be a variable

File: modules/sensor/load_balancer.tf [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+  backend_service        = google_compute_region_backend_service.traffic_ilb_backend_service.id
+  region                 = var.region
+  network                = var.network_prod_name
+  subnetwork             = var.network_mon_subnet_name
+  is_mirroring_collector = true
+  ip_protocol            = "TCP"
```
Comment (thathaneydude) [inline modules/sensor/load_balancer.tf:51]: Should this be UDP?

**Review by thathaneydude — CHANGES_REQUESTED** [review]

File: modules/sensor/load_balancer.tf (hunk 8 above)
Comment (retr0h) [reply modules/sensor/load_balancer.tf:51]: It's not in the training material.

**Review by thathaneydude — APPROVED** [review]

**Pull request merged**